synthtool/gcp/templates/java_library/CONTRIBUTING.md

```diff
@@ -25,4 +25,109 @@ information on using pull requests.
\ No newline at end of file
+[Google's Open Source Community Guidelines](https://opensource.google.com/conduct/).
+
+## Building the project
+
+To build, package and run all unit tests simple run the command
+
+```
+mvn clean verify
+```
+
+### Running Integration tests
+
+To include integration tests when building the project you will need access to
+a GCP Project with a valid service account. 
+
+For instructions on how to generate a service account and corresponding
+credentials JSON see: [Creating a Service Account][1].
+
+Then run the following to build, package, run all unit tests and run all
+integration tests.
+
+```bash
+export GOOGLE_APPLICATION_CREDENTIALS=/path/to/service/account.json
+mvn -Penable-integration-tests clean verify
+```
+
+## Code Samples
+
+Code Samples must be authored in their own Maven modules, and guarded by a
+Maven profile by the name `enable-samples`.
+
+The samples must be separate from the primary project for a few reasons:
+1. Primary projects have a minimum Java version of Java 7 whereas samples have
+   a minimum Java version of Java 8. Due to this we need the ability to
+   selectively exclude samples from a build run.
+2. Many code samples are often dependent on external GCP services and need
+   credentials to access the service.
+3. At this time, code samples are not released as Maven artifacts and must be
+   able to be excluded from any release builds we run.
+   
+### Building
+
+To build and run tests for samples [activate](#profile-activation) the profile
+and run
+
+```bash
+mvn clean verify
+```
+
+Some samples require access to GCP services and require a service account
+
+```bash
+export GOOGLE_APPLICATION_CREDENTIALS=/path/to/service/account.json
+mvn clean verify
+```
+
+### Profile Config
+
+1. To add samples in a profile to your Maven project add the following to your
+`pom.xml`
+
+    ```xml
+    <project>
+      [...]
+      <profiles>
+        <profile>
+          <id>enable-samples</id>
+          <modules>
+            <module>sample</module>
+          </modules>
+        </profile>
+      </profiles>
+      [...]
+    </project>
+    ```
+
+2. [Activate](#profile-activation) the profile.
+3. Define your samples in a normal Maven project in the `sample/` directory
+
+### Profile Activation
+
+To include code samples when building and testing the project you will need to
+enable the `enable-samples` Maven profile.
+
+#### Command line
+
+To activate the Maven profile on the command line add `-Penable-samples` to your
+Maven command.
+
+#### Maven `settings.xml`
+
+To activate the Maven profile in your `~/.m2/settings.xml` add an entry of
+`enable-samples` following the instructions in [Active Profiles][2].
+
+This method has the benefit of applying to all projects you build (and will be
+respected by IntelliJ IDEA) and is recommended if you are going to be
+contributing samples to several projects.
+
+#### IntelliJ IDEA
```

Comment:
This IntelliJ section isn't needed given the above.

Reply:
If the developer picks to add it to their `settings.xml` true, but if they don't this helps them do it (maybe for a single project instead of several).